x-pack/test/security_solution_cypress/cypress/e2e/investigations/timelines/unified_components/query_tab.cy.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  CELL_FILTER_IN_BUTTON,
+  GET_UNIFIED_DATA_GRID_CELL,
+  GET_UNIFIED_DATA_GRID_CELL_HEADER,
+  TIMELINE_DETAILS_FLYOUT,
+  TIMELINE_DETAILS_FLYOUT_CLOSE_BTN,
+} from '../../../../screens/unified_timeline';
+import {
+  TIMELINE_DATA_PROVIDERS_CONTAINER,
+  TIMELINE_FILTER_BADGE,
+} from '../../../../screens/timeline';
+import { GET_DISCOVER_DATA_GRID_CELL_HEADER } from '../../../../screens/discover';
+import { addFieldToTable, removeFieldFromTable } from '../../../../tasks/discover';
+import { login } from '../../../../tasks/login';
+import { visitWithTimeRange } from '../../../../tasks/navigation';
+import { openTimelineUsingToggle } from '../../../../tasks/security_main';
+import { createNewTimeline, executeTimelineSearch } from '../../../../tasks/timeline';
+import { ALERTS_URL } from '../../../../urls/navigation';
+
+describe(
+  'Unsaved Timeline query tab',
+  {
+    tags: ['@ess', '@serverless'],
+    env: {
+      ftrConfig: {
+        kbnServerArgs: [
+          `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+            'unifiedComponentsInTimelineEnabled',
+          ])}`,
+        ],
+      },
+    },
+  },
+  () => {
+    beforeEach(() => {
+      login();
+      visitWithTimeRange(ALERTS_URL);
+      openTimelineUsingToggle();
+      createNewTimeline();
+      executeTimelineSearch('*');
+    });
+    it('should be able to add/remove columns correctly', () => {
+      cy.get(GET_UNIFIED_DATA_GRID_CELL_HEADER('agent.type')).should('not.exist');
+      addFieldToTable('agent.type');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('agent.type')).should('be.visible');
+      removeFieldFromTable('agent.type');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('agent.type')).should('not.exist');
+    });
+
+    context('hover actions', () => {
+      it('should add to timeline successfully', () => {
+        cy.get(GET_UNIFIED_DATA_GRID_CELL('host.name', 0)).then(($el) => {
+          const hostName = $el.text();
+          cy.get(GET_UNIFIED_DATA_GRID_CELL('host.name', 0)).realHover();
+          cy.get(CELL_ADD_TO_TIMELINE_BUTTON).click();
+          cy.get(TIMELINE_DATA_PROVIDERS_CONTAINER).contains(`host.name: "${hostName}"`);
+        });
+      });
+      it('should filter In successfully', () => {
+        cy.get(GET_UNIFIED_DATA_GRID_CELL('user.name', 0)).then(($el) => {
+          const hostName = $el.text();
+          cy.get(GET_UNIFIED_DATA_GRID_CELL('user.name', 0)).realHover();
+          cy.get(CELL_FILTER_IN_BUTTON).click();
+          cy.get(TIMELINE_FILTER_BADGE).should('have.lengthOf', 1);
+          cy.get(TIMELINE_FILTER_BADGE).eq(0).should('have.text', `user.name: ${hostName}`);
+        });
+      });
+    });
+
+    context('flyout', () => {
+      it('should be able to open/close details details/host/user flyout', () => {
+        cy.log('Event Details Flyout');
+        openEventDetailsFlyout(0);
+        cy.get(TIMELINE_DETAILS_FLYOUT_CLOSE_BTN).click();
```

Comment:
All the actions should be wrap in tasks.

Reply:
@MadameSheema , @PhilippeOberti  also pointed out to me that even single statements should be wrapped in tasks. Could you please let me know what is that advantage of that? 

I think `cy.get(....).click()` are much more clear than the task name. isn't it?

Reply:
If anything, I would say re-usability and clarity. It's semantics for the clarity point, but `closeTimelineDetailsFlyout` may be a bit more clear/readable than `cy.get(TIMELINE_DETAILS_FLYOUT_CLOSE_BTN).click();`.

Reply:
+1 to @michaelolo24 explanation. Also because in case of change, let's say that for whatever reason we add an extra step to perform the action, the change in the test can be done in just one place.

Reply:
This is done : 557344da018ff194938c72fcab39726ba7a67e45